core/src/main/java/de/danoeh/antennapod/core/storage/DBWriter.java

```diff
@@ -118,6 +123,26 @@ private static boolean deleteFeedMediaSynchronous(
                 EventBus.getDefault().post(evt);
                 return false;
             }
+
+            deleted = true;
+        } else if (media.getFile_url().startsWith("content://")) {
+            // Local feed
+
+            DocumentFile documentFile = DocumentFile.fromSingleUri(
+                    context,
+                    Uri.parse(media.getFile_url())
+            );
+
+            if (documentFile == null || !documentFile.exists() || !documentFile.delete()) {
+                MessageEvent evt = new MessageEvent(context.getString(R.string.delete_local_failed));
+                EventBus.getDefault().post(evt);
+                return false;
+            }
+
+            deleted = true;
+        }
+
+        if (deleted) {
```

Comment:
I don't think this flag is needed. If it is not deleted, the method already returns, so this part of the code is never reached.